### PR TITLE
feat(LUKE-104): the Conductor key is held by the vault, never the device

### PR DESCRIPTION
**Pull request by dastratakos**
## What

LUKE-104's key half, as Dean asked for it: *"make sure after all of this we aren't storing keys on device and instead are storing in the existing Neon table."* The Conductor key a developer enters is held by Luke's service in its vault (`provider_key` on Neon, migration `0005_crimson_vault.sql`, `POST /api/vault/key`) and by nothing on the Mac. Main as read at `e98c8fed` (#1276, the preferences and vault chains as queues a fiber drains). Rebased onto it as `dcc4c6f7` after #1254 merged as `c3d8854e`, with `1110fad8` on top (a vault step belongs to the era of its press); main has since moved to `67c013d1` (#1274), which merges cleanly with this head and typechecks, lints, and passes the host suite merged. #1267 (the onboarding key step) is stacked on this branch and follows it.

**The finding this replaces.** On main, `credential.setApiKey` encrypts the key with `safeStorage` and writes it into `settings.json` first, then fires the vault POST unawaited, never deletes the local copy, and on success writes the tenant name back into the same file (`compose-settings.ts` → `SettingsStore.setApiKey` → `ProviderKeyVaultSync.keySaved`). The roster, actions, projects, and brain all read Conductor through the vault key on the service; the local key's only jobs were being the thing that got synced and feeding the row's "connected". So every key entered through Settings touched the disk before and after the sync, and stayed there.

### Now

- **The key goes to the vault and nowhere else.** For a cloud agent provider (Conductor is the only one), the `credential.setApiKey` handler takes a new branch, `storeCloudKey`: it requires a signed-in account, applies the same format check the local path did (which already holds the vault's own shape rule: the same 512 cap, and printable ASCII admits no whitespace, pinned by a test at the boundary values), calls `HostedVaultClient.storeKey` (or `deleteKey` for a cleared field) under the account's bearer, and answers the vault's own result. `SettingsStore.setApiKey` is never called with a value for it, so no file, Keychain entry, or cache on the Mac ever sees the key. The OpenAI key for own-key voice has no vault and keeps the local path; the body says so and so does `PRIVACY.md`.
- **Nothing writes the key locally on the way — confirmed on this head.** Every `setApiKey` call site in the tree, tests aside: the store's own method; the awaited shim over it; the local path in the `credential.setApiKey` handler, which sits *after* the `isCloudAgentProviderId` early return, so a Conductor key never reaches it; and two calls with `undefined` (the migration's and the store's deletion of a leftover an earlier build kept). `ProviderKeyVaultSync` is deleted, so nothing writes the tenant record either.
- **"Connected" is read from the vault's list.** The settings composer holds the set of providers the vault lists a key for (`GET /api/vault/keys`), refreshed when the account's capabilities open, moved by each store and delete, and emptied at sign-out. `SettingsStore` takes it as a required seam (`vaultKeyHeld`) and, for a cloud provider, reports a new `CREDENTIAL_SOURCE.SERVICE` or `NONE` from it, never from the file or the environment. The row says "Held by Luke's service" and offers Delete for it.
- **One migration, then nothing local.** A Conductor key an earlier build kept encrypted here is handed to the vault once when the account's capabilities open, only for the account the earlier build's tenant record names, and deleted from the Mac when the vault confirms; a vault that refuses leaves it for the next sign-in rather than losing the developer's key. An environment `CONDUCTOR_API_KEY` is not Luke's to send and no longer connects anything on the row.
- **A vault step belongs to the era of its press, and its calls carry only that account's bearer** (answering Cursor's security review on `3406e8c0`, and Bugbot's and the security review's finding on `dcc4c6f7`). A step is offered with the generation standing and the account signed in at the press, read before the offer, and begins only if that era still stands when the queue reaches it (`vaultStepEraStands`); a Save that waited behind a reconcile's round trips while the account signed out and another signed in does nothing and answers a refusal. The vault client's bearer is the era's account's access token or nothing (`vaultStepBearer`), so a call that reached the wire under another account reads no credential and never travels; the 401 retry re-reads the same way. Both rules have tests on their values.
- **The sync switch and tenant record are gone**, as LUKE-104 says: the `syncProviderKeys` setting, its `VAULT_SYNC` side effect on both tables, the Connections page's Sync section, the `setVaultSyncAccount` store method and its awaited face, and `ProviderKeyVaultSync` itself. The `vaultSyncAccount` field is read-only now, kept for the migration's tenant match; a stale `syncProviderKeys` field in an existing `settings.json` is ignored by the reader, which iterates the schema.

### The rebase

Main moved twice under this PR. First the settings store and composer onto Effect (#1270, #1271, #1272), then #1276 turned the composer's two promise chains into Effect `Queue`s drained by fibers of its lifetime scope, with every step the store's own effect and `emitSettings` an effect. The rebase keeps that shape: `SettingsStore.snapshot` reads the vault seam inside its `Effect.forEach`, `readVaultSyncAccount` stays an effect, and the vault chain is a `Queue` drained by `drainVaultActions` beside the preferences queue. The reconcile is an offer, so `openCapabilities` waits on no network; the save is a step answered through a `Deferred`, so the `credential.setApiKey` handler yields the vault's own answer once the steps ahead of it are done. The two follow-up fixes from the earlier heads (`c29424f1`, the bearer bound to the step's account; `89b4572c`, the refusal that names its reason) are folded into the one commit, since #1276 rewrote the chain both sat in. `AGENTS.md` and the ADR are untouched: the ADR's line that "the settings composer's own account-preferences and provider-key-vault chains are `Queue`s drained by a fiber of the composer's lifetime scope, and every step either takes is the store's own effect" stays true of the new chain.

### Documents

- **`AGENTS.md`: no change.** #1251 pruned every agent guide to rules a tool does not already enforce and cut product specification from them by policy, so the vault rule lives in this description, in `PRIVACY.md`, and in the code's own comments. `git diff origin/main -- AGENTS.md` is empty on this head.
- **`PRIVACY.md`** is rewritten in as many words: the key is sent in the same press and nothing of it is written to the Mac; the row reads from the vault's list; deleting removes it there; the one-time migration, and only for the account the earlier build's tenant record names; the environment key; the OpenAI key as the exception. The summary and "Your choices" sentences follow.
- The guide's Cloud providers fact says the same, so Luke describes it correctly when asked.

### LUKE-104

This covers the ticket's key half ("enter the Conductor key straight into the vault; the desktop holds no provider keys; the Sync provider keys switch and tenant record are no longer drawn"). Its other bullet, a fixture service client for roster, messages, and asks, is not touched here: fixture runs have no account, so the vault branch refuses before anything travels, and the fixture roster is already the service client's fixture path from the E lane. **LUKE-104 closes against this PR** for the key half, with that bullet noted on the ticket.

## Tests

Values and structure: a cloud provider's source is the vault's answer whatever the Mac holds or reads (held → `SERVICE`, not held → `NONE`, an environment key notwithstanding, and `NONE` signed out however the set reads); the local-storage tests move to the OpenAI key, which is what the file still stores; the host refuses a Conductor key signed out with a reason and a snapshot whose Conductor source is `none`; a vault call's bearer is the step's account's token, `undefined` for another account, signed out, or outside a step; the schema, guide, and desktop suites pass with the setting gone.

## Verification

- **`dcc4c6f7`, in the Linux cloud sandbox:** `./scripts/check.sh` exit 0 — knip, `pnpm run lint` (Biome and oxlint), `pnpm --recursive run typecheck`, and all four test shards passed (1055, 832, 926, 878 tests; one skipped), with no `[vitest-worker]: Timeout calling onTaskUpdate`, so LUKE-187 is not in play on this run. On this head, `1110fad8`: `pnpm run lint` clean, every package's typecheck clean, and the whole suite 463 files / 3694 tests passing (one skipped). Host suite alone on `dcc4c6f7`: 44 files / 349 tests, including the store's vault-source test, the host's signed-out refusal through the new queue, the bearer rule's values, and the door's subsumption of the vault's shape at the boundary values.
- **The two rebases.** First onto `c66b7dae` as `438c48a4` (`git range-diff ec93417b..81dd244e c66b7dae..438c48a4`: the three #1254 commits drop, the one PR commit carries), where the conflicts were the Effect moves in `compose-settings.ts`, `settings-store.ts`, and its test; then onto `e98c8fed` as `dcc4c6f7`, where #1276 had rewritten the chain and the conflicts were `compose-settings.ts`, `compose-host.ts`, `settings-side-effects.ts`, and the deleted `provider-key-vault-sync.ts`, resolved as described above. The oxlint edge rule (`anti-slop/no-run-promise-outside-edges`) shaped the first port; the second yields the store's own effects throughout, so no promise face is involved.
- **Earlier heads.** The first `./scripts/check.sh` on this branch failed lint on a `typeof` assertion in the new host test (`anti-slop/no-runtime-typeof`), fixed as `isWireString` in `3536a628`. Bugbot's ten findings across `9edf6659`…`fb54887b` were fixed in `3536a628`, `eee2dfaf`, `227241da`, `fb54887b`, and `81dd244e`, each answered on its thread; Cursor's security review on `3406e8c0` (a save begun under one account could POST under another's bearer) was fixed in `c29424f1`, and Bugbot's one finding on `c29424f1` (a cloud key's refusal blamed spaces for any shape failure) in `89b4572c`, both answered on their threads and both carried into `dcc4c6f7`. On `89b4572c` all six required contexts and both Cursor verdicts were green before main moved. On `dcc4c6f7` the six contexts and the security review were green and Bugbot and the security review each raised the same finding (a Save waiting on the queue could run under a later sign-in), fixed in `1110fad8` on their threads.
- **Not verified, and CI cannot:** built in a Linux cloud sandbox with no macOS and no account, so no key was stored against the real vault from a desktop here, and no Mac ran the migration of a leftover key. `./scripts/verify.sh` did not run and there is no visual evidence; the drawn change is the Sync section leaving the Connections page and the Conductor row's new status words. There is no macOS job in CI, so those remain the reviewer's own. The service side (`POST`/`GET`/`DELETE /api/vault/key(s)`, `provider_key` on Neon) is unchanged.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/host/src/compose-settings.ts [hunk 1]
```diff
@@ -160,31 +184,87 @@ export const composeSettings = (): Effect.Effect<
...
+    async function reconcileVaultKeys(): Promise<void> {
+      await migrateLocalCloudKeys();
+      await refreshVaultKeys();
+      await emitSettings();
+    }
+
+    function forgetVaultKeys(): void {
+      if (vaultKeys.size === 0) return;
+      vaultKeys.clear();
+      void emitSettings();
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in 3536a628: every touch of the vault's key list now rides one chain (`enqueueVault`), so a refresh, a store, a delete, and the forget at sign-out run in the order they were asked and never interleave.

File: apps/desktop/src/renderer/settings/connection-schema.tsx [hunk 2]
```diff
@@ -275,21 +276,26 @@ function credentialConnection(
...
-      // typed in, so both connected states offer the same editor and only the
-      // unconnected one is asked to connect.
-      const stored = source(input) === CREDENTIAL_SOURCE.ENCRYPTED_FILE;
+      // Deleting is only ever for a key Luke keeps, here or with his service;
+      // one read from the environment is not Luke's to remove. Either can be
+      // superseded by a key typed in, so both connected states offer the same
+      // editor and only the unconnected one is asked to connect.
+      const stored =
+        source(input) === CREDENTIAL_SOURCE.ENCRYPTED_FILE ||
+        source(input) === CREDENTIAL_SOURCE.SERVICE;
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply apps/desktop/src/renderer/settings/connection-schema.tsx:298]: Fixed in 3536a628: a refused Keychain now blocks only a key kept on this Mac; a cloud provider's Connect and Replace are not gated on `storageUnavailable`, since that path writes nothing locally.

File: packages/host/src/compose-settings.ts [hunk 3]
```diff
@@ -160,31 +184,87 @@ export const composeSettings = (): Effect.Effect<
...
+        const stored = await hostedVault.storeKey(providerId, local);
+        if (stored?.stored) await store.setApiKey(providerId, undefined);
+      }
+    }
+
+    async function reconcileVaultKeys(): Promise<void> {
+      await migrateLocalCloudKeys();
+      await refreshVaultKeys();
+      await emitSettings();
+    }
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in 3536a628: every step that awaited the service re-reads the signed-in account before it writes `vaultKeys` (the list, the migration's delete, the store, the delete), and installs nothing when the account changed or is gone; the forget is queued behind whatever is in flight.

File: packages/host/src/compose-settings.ts [hunk 4]
```diff
@@ -160,31 +184,87 @@ export const composeSettings = (): Effect.Effect<
...
+     * not Luke's to send and is left where the shell put it.
+     */
+    async function migrateLocalCloudKeys(): Promise<void> {
+      for (const providerId of Object.values(CLOUD_AGENT_PROVIDER_ID)) {
+        const local = await store.readStoredApiKey(providerId);
+        if (local === undefined) continue;
+        const stored = await hostedVault.storeKey(providerId, local);
+        if (stored?.stored) await store.setApiKey(providerId, undefined);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in 3536a628: the migration reads the tenant record the earlier build kept (`vaultSyncAccount`, now read-only and never written) and hands a leftover key to the vault only when that record names the signed-in account by id or email; a key with no tenant, or another account's, is sent nowhere and stays where it is. PRIVACY.md and AGENTS.md say so.

**Comment by dastratakos** [comment]
Full `./scripts/check.sh` on `3536a628`: repository checks, knip, Biome and oxlint, every package's typecheck, and the build passed; all 456 test files passed, and the run exited 1 only on one `[vitest-worker]: Timeout calling "onTaskUpdate"` naming no file, which is LUKE-187.

File: packages/host/src/compose-settings.ts [hunk 5]
```diff
@@ -160,31 +184,136 @@ export const composeSettings = (): Effect.Effect<
...
+      vaultActions = next.then(
+        () => undefined,
+        () => undefined,
+      );
+      return next;
+    }
+
+    async function signedInAccountKey(): Promise<string | undefined> {
+      const account = await store.readAccount();
+      return account ? (account.id ?? account.email) : undefined;
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in eee2dfaf: the in-flight checks compare the account by address alone, which a refresh cannot add or drop, so a store, list, or local delete that outlives an identity refresh is no longer read as a sign-out; the migration's tenant match still accepts either name.

File: packages/host/src/compose-host.ts [hunk 6]
```diff
@@ -212,6 +212,7 @@ export const hostAssemblyLayer: Layer.Layer<
       yield* capabilities.disarm;
       conversation.reset();
       observation.stopObservation();
+      settings.forgetVaultKeys();
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-host.ts:218]: Addressed in 227241da. Sign-out: `forgetVaultKeys` empties the set at once rather than behind in-flight vault calls, and the sign-out's own emit reads the row as not connected; every in-flight call already re-reads the account before it writes, so none can refill the set. Sign-in: the first list is asked for once more when it cannot be read, and the emit that follows the list is what brings the row to connected. The moment between the sign-in's first emit and the list landing still reads not connected, which is the honest state: the row says what the vault last listed, and until it has been asked it has listed nothing.

File: packages/host/src/compose-settings.ts [hunk 7]
```diff
@@ -160,31 +184,148 @@ export const composeSettings = (): Effect.Effect<
...
+    /**
+     * The account is gone, so the set empties now rather than behind whatever
+     * vault call is still in flight: every such call re-reads the account
+     * before it writes, so nothing it answers can refill the set for an
+     * account that has signed out, and the sign-out's own emit already reads
+     * the row as not connected.
+     */
+    function forgetVaultKeys(): void {
+      vaultKeys.clear();
+    }
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts:454]: Fixed in fb54887b. The row is tied to the account inside the same snapshot: a cloud provider reads `service` only while the store's own persisted account stands, so the sign-out's first emit already says not connected whatever the set holds. In-flight vault calls are fenced by a generation `forgetVaultKeys` moves as well as by the account re-read, so a list or store that returns after the sign-out installs nothing however the settings file looked mid-clear.

File: packages/host/src/compose-settings.ts [hunk 8]
```diff
@@ -160,31 +184,148 @@ export const composeSettings = (): Effect.Effect<
...
+      if (tenant === undefined || (tenant !== account.id && tenant !== account.email)) return;
+      for (const providerId of Object.values(CLOUD_AGENT_PROVIDER_ID)) {
+        const local = await store.readStoredApiKey(providerId);
+        if (local === undefined) continue;
+        const stored = await hostedVault.storeKey(providerId, local);
+        if (!stored?.stored) continue;
+        if ((await signedInAccountKey()) !== account.email) return;
+        await store.setApiKey(providerId, undefined);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in fb54887b. The migration reads the vault's list first and uploads a leftover only for a provider the vault holds no key for; where the vault already holds one, the vault's key stands and the leftover is deleted here without travelling. `reconcileVaultKeys` is now the first thing on the vault's chain when capabilities open, ahead of any save the developer makes meanwhile, which then wins as the newest word on the same chain; and `storeCloudKey` deletes a leftover local copy in the same press.

File: packages/host/src/settings-store.ts [hunk 9]
```diff
@@ -652,16 +663,30 @@ export class SettingsStore {
...
+        // the vault, and a key this Mac still holds or reads from its shell
+        // is not one Luke observes with.
+        if (isCloudAgentProviderId(provider.id)) {
+          // Held only while an account stands in this same snapshot: the
+          // vault's list is the account's, so a sign-out reads not connected
+          // in the very emit that reports it, whatever the list last said.
+          const source =
+            persisted.account !== undefined && this.#vaultKeyHeld(provider.id)
+              ? CREDENTIAL_SOURCE.SERVICE
+              : CREDENTIAL_SOURCE.NONE;
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/settings-store.ts]: Fixed in 81dd244e: the account composer empties the vault set on the sign-out edge itself, inside the change that reports the departure and ahead of its own settings emit, so the very snapshot that says signed out reads every cloud provider as not connected, whatever the store still holds at that instant.

File: packages/host/src/compose-settings.ts [hunk 10]
```diff
@@ -160,31 +184,180 @@ export const composeSettings = (): Effect.Effect<
...
+        const generation = vaultGeneration;
+        const account = await store.readAccount();
+        if (!account) return;
+        const held = await refreshVaultKeys(generation, account.email);
+        if (held === undefined) return;
+        if (await migrateLocalCloudKeys(generation, account, held)) {
+          await refreshVaultKeys(generation, account.email);
+        }
+        if (!(await vaultStillCurrent(generation, account.email))) return;
+        await emitSettings();
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in 81dd244e: once the vault has answered that it stored a migrated key, the provider is added to the set at once, before the leftover is deleted here, so the row says held from that moment whether or not the list that follows can be read.

**Comment by dastratakos** [comment]
**Orchestrator:** #1254 (your base) MERGED as `c3d8854e`. **Retarget this PR to `main`, rebase onto the squash, answer the one unresolved thread (fix with a test or dispute with the reason, resolve it yourself naming the commit), get fresh bot verdicts, report the head here.** #1267 follows this one the same way. LUKE-104 closes against this PR — say so in the body. Do not arm a watcher.

**Comment by dastratakos** [comment]
Orchestrator takeover: Conductor session `conductor/t-1260-take-over-1260` (T-1260) now owns this branch. Original worker: please stand down.

File: packages/host/src/compose-settings.ts [hunk 11]
```diff
@@ -160,31 +184,139 @@ export const composeSettings = (): Effect.Effect<
...
+            "Sign in first: this key is held by Luke's service, not on this Mac.",
+          );
+        }
+        const normalized = apiKey?.trim();
+        if (normalized) {
+          const rejection =
+            apiKeyRejection(normalized, CREDENTIAL_PROVIDERS[providerId].keyFormat) ??
+            (vaultKeyIsStorable(normalized) ? undefined : "That API key contains spaces.");
+          if (rejection) return refusedSettings(rejection);
+          const stored = await hostedVault.storeKey(providerId, normalized);
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in c29424f1, after the rebase onto main as 438c48a4. Confirmed against the head: `storeCloudKey` and the migration read the account at the head of their step, `HostedVaultClient` reads its bearer fresh at request time through `accountBearer`, and the account session is not on `enqueueVault`'s chain, so a store or migration that began under one account could reach the wire under a later account's bearer; the re-read after the call only refused the local set. Now the composer records, for the step on the chain, the address it began under (`beginVaultStep`), and the vault client's `readAccessToken` answers through `vaultStepBearer`: that account's token while it is the one signed in at the moment of the request, otherwise nothing, so the call reads no credential, never travels, and the step answers a refusal. The 401 retry re-reads the same way, and between steps no account is recorded, so a call outside the chain travels as nobody. `vault-step-bearer.test.ts` asserts the values: the step's own account's token, `undefined` for another account, signed out, or outside a step.

File: packages/host/src/compose-settings.ts [hunk 12]
```diff
@@ -179,31 +213,213 @@ export const composeSettings = (): Effect.Effect<
...
+        if (accountKey === undefined) {
+          return refusedCloudKey(
+            "Sign in first: this key is held by Luke's service, not on this Mac.",
+          );
+        }
+        const normalized = apiKey?.trim();
+        if (normalized) {
+          const rejection =
+            apiKeyRejection(normalized, CREDENTIAL_PROVIDERS[providerId].keyFormat) ??
+            (vaultKeyIsStorable(normalized) ? undefined : "That API key contains spaces.");
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in 89b4572c. Confirmed against the head, and the branch was also unreachable: `apiKeyRejection` runs first, caps at 512 (the vault's own `VAULT_KEY_MAX_LENGTH`), and admits printable ASCII alone, which has no whitespace, so a key it passes is one `vaultKeyIsStorable` stores, and a key it refuses already names its reason (too long, unsupported characters). The mislabeled fallback is gone; the cloud path holds the key to the one door, with a comment saying why that door already holds the vault's rule. A test in `settings-store.test.ts` pins the subsumption at the boundary values: 512 characters pass both, 513 is "too long" and not storable, and a key with spaces, tabs, or newlines is "unsupported characters" and not storable.

**Comment by dastratakos** [comment]
**Orchestrator:** this went CONFLICTING at 18:29Z because **#1276** (`e98c8fed`, another lane: *the preferences and vault chains as queues a fiber drains*) landed on the vault sync you build on — the same seam. Rebase onto `origin/main` once more; the conflict is in the vault/settings chain, not in your intent. Keep the rule: the key goes to the hosted vault, never to disk. Report READY here when green with zero threads; I press this one.

File: packages/host/src/compose-settings.ts (hunk 11 above)
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Carried into dcc4c6f7 after the rebase onto #1276 (`e98c8fed`), which rewrote the chain this fix sits in: the vault chain is now an Effect `Queue` drained by a fiber of the composer's lifetime scope, and the bearer rule is unchanged — `beginVaultStep` records the step's account, `vaultStepBearer` answers that account's token or nothing at request time, and `vault-step-bearer.test.ts` still asserts the values.

File: packages/host/src/compose-settings.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Carried into dcc4c6f7 after the rebase onto #1276 (`e98c8fed`): the cloud path still holds the key to `apiKeyRejection` alone, and the subsumption test in `settings-store.test.ts` is unchanged.

File: packages/host/src/compose-settings.ts [hunk 13]
```diff
@@ -246,9 +265,246 @@ export const composeSettings = (): Effect.Effect<
...
+            // printable ASCII admits no whitespace, so a key that passes here is
+            // one the vault stores, and each refusal names its own reason.
+            const rejection = apiKeyRejection(
+              normalized,
+              CREDENTIAL_PROVIDERS[providerId].keyFormat,
+            );
+            if (rejection) return yield* refusedSettings(rejection);
+            const stored = yield* Effect.promise(() =>
+              hostedVault.storeKey(providerId, normalized),
+            );
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts:522]: Fixed in 1110fad8. Confirmed against the head: the step read the account and generation when the queue reached it, and a Save waits behind a reconcile's round trips, so a sign-out and another sign-in in that window would have run the step under the later account. Now a step is offered with the era of its press — the generation standing and the account signed in, read before the offer (`vaultEraNow`) — and begins only if that era still stands when the queue reaches it (`beginVaultStep` over `vaultStepEraStands`); otherwise it does nothing and answers a refusal. A sign-out moves the generation and a direct account change moves the address, so neither lets a waiting Save begin under another account, and the bearer is bound to the era's account, so its calls travel as the account that pressed or not at all. The reconcile is offered with its own era the same way. `vault-step-era.test.ts` asserts the values: the same era stands; a moved generation does not, whoever signed in since; another account in the same generation does not.

File: packages/host/src/compose-settings.ts [hunk 14]
```diff
@@ -246,9 +265,246 @@ export const composeSettings = (): Effect.Effect<
...
+     */
+    const storeCloudKey = (
+      providerId: CloudAgentProviderId,
+      apiKey: string | undefined,
+      reporter: string | undefined,
+    ): Effect.Effect<SettingsUpdateResult> =>
+      enqueueVault(
+        Effect.gen(function* () {
+          const generation = vaultGeneration;
+          const accountKey = (yield* beginVaultStep())?.email;
```
[the remark this answers is not in the document]
Comment (dastratakos) [reply packages/host/src/compose-settings.ts]: Fixed in 1110fad8. Confirmed against the head: the step read the account and generation when the queue reached it, and a Save waits behind a reconcile's round trips, so a sign-out and another sign-in in that window would have run the step under the later account. Now a step is offered with the era of its press — the generation standing and the account signed in, read before the offer (`vaultEraNow`) — and begins only if that era still stands when the queue reaches it (`beginVaultStep` over `vaultStepEraStands`); otherwise it does nothing and answers a refusal. A sign-out moves the generation and a direct account change moves the address, so neither lets a waiting Save begin under another account, and the bearer is bound to the era's account, so its calls travel as the account that pressed or not at all. The reconcile is offered with its own era the same way. `vault-step-era.test.ts` asserts the values: the same era stands; a moved generation does not, whoever signed in since; another account in the same generation does not.

**Comment by dastratakos** [comment]
**Orchestrator:** gate verified on `1110fad8` — CLEAN, both Cursor verdicts SUCCESS, zero unresolved threads, body states the key goes to the hosted vault and never to disk, LUKE-104 closes against this, PRIVACY.md moves with it. The only non-green context is `Preview shape`, inert by design. **Pressed.**

**Comment by dastratakos** [comment]
**READY** — head `1110fad8`, base `main`, not enqueued (orchestrator-word class: this PR rewrites `PRIVACY.md`'s credential paragraph; `AGENTS.md` is unchanged after #1251). Reported by Conductor session `conductor/t-1260-take-over-1260` (T-1260) at 2026-09-12T19:04Z.

**Gate on `1110fad8`**
- Required contexts: TypeScript checks, Postgres migrations and store, Tests 1/4–4/4 — all `success`.
- Cursor Bugbot: `success` ("no issues found"). Cursor Security Reviewer: `success`.
- Review threads: 0 unresolved of 14; every one re-read against its head and resolved by me naming the commit (`3536a628`, `eee2dfaf`, `227241da`, `fb54887b`, `81dd244e` from before the takeover; `c29424f1` and `89b4572c` carried into `dcc4c6f7`; `1110fad8` for the two threads on `dcc4c6f7`).
- `mergeStateStatus: CLEAN`, `mergeable: MERGEABLE`. Main as read at `67c013d1` (#1274): two commits past this branch's base `e98c8fed`; a real merge of the two typechecks every package, lints clean, and passes the host suite, and the ruleset does not require the head to be up to date. Squash is the queue's merge method.

**What changed under the takeover**
- Rebased twice: onto `c66b7dae` (dropping #1254's three commits, merged as `c3d8854e`), then onto `e98c8fed` after #1276 rewrote the composer's chains as fiber-drained `Queue`s. The vault chain now rides that shape: a `Queue` drained by `drainVaultActions` in the composer's lifetime scope, every step the store's own effect, the reconcile an offer, the save a step answered through a `Deferred`.
- Cursor's security review (`3406e8c0`): the bearer of every vault call is bound to the account of the step (`vaultStepBearer`, tested), so a call reaching the wire under another account reads no credential and never travels.
- Bugbot on `c29424f1`: the mislabeled spaces refusal was a dead branch; `apiKeyRejection` already caps at the vault's 512 and admits no whitespace, pinned by a boundary-value test.
- Bugbot and the security review on `dcc4c6f7`: a Save waiting on the queue could have begun under a later sign-in. A step is now offered with the era of its press (generation and account, `vaultStepEraStands`, tested) and begins only if it still stands.

**Dean's rule, confirmed on this head:** the Conductor key goes to the hosted vault (`provider_key` on Neon, `POST /api/vault/key`) and nowhere on the device. `ProviderKeyVaultSync` is deleted; every `setApiKey` call site is the store's own method, its awaited face, the local path in the `credential.setApiKey` handler (after the `isCloudAgentProviderId` early return, so a Conductor key never reaches it), and two deletions with `undefined`. The row's "connected" reads from the vault's list. `PRIVACY.md` says so in as many words.

**LUKE-104 closes against this PR** for its key half; the fixture-service-client bullet is noted on the ticket as not in this PR.

**Not verified, and CI cannot:** no macOS job and no account in the sandbox, so no key was stored against the real vault from a desktop, no Mac ran the leftover-key migration, and `./scripts/verify.sh` produced no visual evidence for the Connections page change. `./scripts/check.sh` exit 0 on `dcc4c6f7`; on `1110fad8` lint, every typecheck, and the whole suite (463 files / 3694 tests) pass locally, and CI agrees. No LUKE-187 worker timeout in either run.

**#1267** is stacked on `vault/key-entry`, which has been force-pushed twice; it needs a rebase onto this head (or onto main after the squash) before it follows.
